### PR TITLE
CI Improvements

### LINKED ISSUE #88: Linter Version Check is Broken
**Describe the bug**

When running `make check` using [golangci/golangci-lint:v1.40.1](https://hub.docker.com/r/golangci/golangci-lint), a version difference is reported even though the correct version is installed.

**To Reproduce**

Run `make check` inside a `golangci/golangci-lint:v1.40.1` container ([example](https://app.circleci.com/pipelines/github/sylabs/singularity/170/workflows/905b4a34-a58a-4e69-b8af-cacf2b714989/jobs/1520/parallel-runs/0/steps/0-105)):

```sh
$ make -C ./builddir check
...
 CHECK golangci-lint
cd /root/project && \
	scripts/run-linter run --verbose --build-tags "containers_image_openpgp sylog oci_engine singularity_engine fakeroot_engine apparmor selinux seccomp" ./...
W: 
W: ********************************************************************************************
W: Singularity's CI checks will use version 1.40.1 of golangci-lint.
W: Your installed version differs. Please use 1.40.1 if you experience issues.
I: 
I: The output of "golangci-lint version" is:
I: 
I:     golangci-lint has version v1.40.1 built from 625445b1 on 2021-05-14T11:59:47Z
I: 
I: It was found in the following location:
I: 
I:     /usr/bin/golangci-lint
I: ********************************************************************************************
I: 
INFO [config_reader] Config search paths: [./ /root/project /root /]
...
```

**Expected behavior**

`make check` runs without warnings when the correct version of `golangci-lint` is installed.

**Installation Method**

Source

**Additional context**

Bug does not occur when installed using the [install script](https://golangci-lint.run/usage/install/#local-installation) locally.

### PR DESCRIPTION
**Pull request by tri-adam**
Add `lint-source` job, removing duplicate linting previously executed in `check-*` jobs. Rename `check-*` jobs to `build-source-*` to reflect new semantics. Remove `run-linter` script to fix #88. Set build tags explicitly in linter configuration file to ensure consistent linting across platforms. Fix lint in `sylog` package.

### CODE REVIEW

**Review by dtrudg — APPROVED** [review]

**Comment by tri-adam** [comment]
Converting to draft, will hold off on this until #99 is merged.

**Review by dtrudg — APPROVED** [review]
LGTM

**Pull request merged**
